python/ray/util/concurrent/futures/ray_executor.py

```diff
@@ -0,0 +1,658 @@
+        """
+        self.pool = [i for i in self.pool if i != pool_actor]
+        pool_actor["actor"].exit.remote()
+        return pool_actor["actor"]
+
+
+class _RandomActorPool(_AbstractActorPool):
+
+    """This class manages a pool of Ray actors by distributing tasks amongst
+    them in a random choice fashion. Functions are executed remotely in the
+    actor pool using submit().
+
+    ...
+
+    Attributes
+    -----------
+    num_actors : int
+        Specify the size of the actor pool to create.
+    initializer : Callable
+        A function that will be called remotely in the actor context before the
+        submitted task (for compatibility with
+        concurrent.futures.ThreadPoolExecutor).
+    initargs : tuple
+        Arguments for initializer (for compatibility with
+        concurrent.futures.ThreadPoolExecutor).
+    max_tasks_per_actor : int
+        The maximum number of tasks to be performed by an actor before it is
+        gracefully killed and replaced (for compatibility with
+        concurrent.futures.ProcessPoolExecutor).
+    """
+
+    def next(self) -> _PoolActor:
+        """
+        Get the next priority member of the actor pool
+
+        Returns
+        -------
+        _PoolActor
+            The current priority actor in the pool
+        """
+        return random.choice(self.pool)
+
+
+class _RoundRobinActorPool(_AbstractActorPool):
+
+    """This class manages a pool of Ray actors by distributing tasks amongst
+    them in a simple round-robin fashion. Functions are executed remotely in
+    the actor pool using submit().
+
+    ...
+
+    Attributes
+    -----------
+    num_actors : int
+        Specify the size of the actor pool to create.
+    initializer : Callable
+        A function that will be called remotely in the actor context before the
+        submitted task (for compatibility with
+        concurrent.futures.ThreadPoolExecutor).
+    initargs : tuple
+        Arguments for initializer (for compatibility with
+        concurrent.futures.ThreadPoolExecutor).
+    max_tasks_per_actor : int
+        The maximum number of tasks to be performed by an actor before it is
+        gracefully killed and replaced (for compatibility with
+        concurrent.futures.ProcessPoolExecutor).
+    """
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.index: int = 0
+        return
+
+    def next(self) -> _PoolActor:
+        """
+        Get the next indexed member of the actor pool
+
+        Returns
+        -------
+        _PoolActor
+            The current priority actor in the pool
+        """
+        obj = self.pool[self.index]
+        self.index += 1
+        if self.index >= len(self.pool):
+            self.index = 0
+        return obj
+
+
+@PublicAPI(stability="alpha")
+class RayExecutor(Executor):
+
+    """RayExecutor is a drop-in replacement for ProcessPoolExecutor and
```

Comment:
Good job on the docstring. Let's also add important information about thread safety and error handling.